### ISSUE TITLE
feature: add more meaningful fields to crd objects

### ISSUE THREAD

**Issue by googs1025** [kubernetes-sigs/lws#210]
**What would you like to be added**:
Currently, when using kubectl get with lws , there are only two fields displayed: name and age. We should provide more meaningful fields to make it more convenient to use.
```bash
root@VM-0-8-ubuntu:/home/ubuntu# kubectl get lws
NAME                        AGE
leaderworkerset-sample      5h21m
```

We should provide more meaningful fields, which will be more convenient to use
as follows:
```bash
root@VM-0-8-ubuntu:/home/ubuntu# kubectl get lws
NAME                     READY   UP-TO-DATE   AGE
leaderworkerset-sample   3/3     3            5h21m
```

There were some minor problems during implementation. FRY: https://github.com/kubernetes-sigs/kubebuilder/discussions/4134
I am not sure whether the current JSONPath of kubebuilder can achieve a display effect similar to (X/Y). In addition, should we add other useful fields (maybe we can also discuss this PR)? If not, I will close this issue

**Why is this needed**:
- provide more meaningful fields
**Completion requirements**:

This enhancement requires the following artifacts:

- [ ] Design doc
- [ ] API change
- [ ] Docs update

The artifacts should be linked in subsequent comments.

**Comment by liurupeng** [comment]
@googs1025 Yes, I think adding "READY" and "UP-TO-Update" will be very useful

**Comment by googs1025** [comment]
That's great, but there's a small problem. https://github.com/kubernetes-sigs/kubebuilder/discussions/4134  This problem can be solved by adding an extra field to the status, but I don't think this is a good way to handle it. Maybe we can wait until this problem is solved before implementing it

**Comment by googs1025** [comment]
issue tracked: https://github.com/kubernetes-sigs/controller-tools/issues/1051

**Comment by kerthcet** [comment]
Glad to see this aligned with deployment or statefulset which has richer columns.

**Comment by ahg-g** [comment]
@googs1025 are you still planning to address this issue?

**Comment by googs1025** [comment]
thanks for reply. @ahg-g 
Yes, but there are currently two ways to implement it. 🤔 One is to add a new field in the API to store similar fields. The other method requires the support of upstream k8s, issue link: https://github.com/kubernetes/kubernetes/issues/127246  kep: https://github.com/kubernetes/enhancements/issues/4595

**Comment by nayihz** [comment]
Do we have to display the `Ready` field like this?
```
root@VM-0-8-ubuntu:/home/ubuntu# kubectl get lws
NAME                     READY   UP-TO-DATE   AGE
leaderworkerset-sample   3/3     3            5h21m
```
If so, maybe we should add a status field to record the ready Replicas and total replicas together. (I don't think it's necessary)
If not, we only need to add some markers to display what we want. Like this:
```
// +kubebuilder:printcolumn:name="Ready",type=string,JSONPath=`.status.ready`
```

**Comment by xigang** [comment]
Fxies: https://github.com/kubernetes-sigs/lws/pull/580
